### PR TITLE
feat: Clear cache storage on logout (WEBAPP-6383)

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: src/script/cache/CacheRepository.ts [hunk 1]
```diff
@@ -29,18 +30,20 @@ import {StorageKey} from '../storage/StorageKey';
...
   }
 
   /**
    * Deletes cached data.
    *
    * @param {boolean} [keepConversationInput=false] - Should conversation input be kept
-   * @param {Array<string>} [protectedKeyPatterns=[StorageKey.AUTH.SHOW_LOGIN]] - Keys which should NOT be deleted from the cache
+   * @param {Array<string>} [protectedKeyPatterns=[StorageKey.AUTH.SHOW_LOGIN]] - Keys which should NOT be deleted from
+   *   the cache
    * @returns {Array<string>} Keys which have been deleted from the cache
```
Comment (ffflorian) [inline src/script/cache/CacheRepository.ts]: Type annotations in JSDoc are not needed in TypeScript.

File: src/script/cache/CacheRepository.ts [hunk 2]
```diff
@@ -17,30 +17,30 @@
...
   }
 
   /**
    * Deletes cached data.
    *
-   * @param {boolean} [keepConversationInput=false] - Should conversation input be kept
-   * @param {Array<string>} [protectedKeyPatterns=[StorageKey.AUTH.SHOW_LOGIN]] - Keys which should NOT be deleted from the cache
-   * @returns {Array<string>} Keys which have been deleted from the cache
+   * @param [keepConversationInput=false] - Should conversation input be kept
+   * @param [protectedKeyPatterns=[StorageKey.AUTH.SHOW_LOGIN]] - Keys which should NOT be deleted from
```
Comment (ffflorian) [inline src/script/cache/CacheRepository.ts]:
```suggestion
   * @param protectedKeyPatterns - Keys which should NOT be deleted from
```

File: src/script/cache/CacheRepository.ts [hunk 3]
```diff
@@ -17,30 +17,30 @@
...
+    return keyList;
   }
 
   /**
    * Deletes cached data.
    *
-   * @param {boolean} [keepConversationInput=false] - Should conversation input be kept
-   * @param {Array<string>} [protectedKeyPatterns=[StorageKey.AUTH.SHOW_LOGIN]] - Keys which should NOT be deleted from the cache
-   * @returns {Array<string>} Keys which have been deleted from the cache
+   * @param [keepConversationInput=false] - Should conversation input be kept
```
Comment (ffflorian) [inline src/script/cache/CacheRepository.ts]:
```suggestion
   * @param keepConversationInput - Should conversation input be kept
```

File: src/script/config/dependenciesGraph.js [hunk 4]
```diff
@@ -81,7 +79,6 @@ export {
   BackendClient,
   BackupService,
   BroadcastService,
-  CacheRepository,
```
Comment (ffflorian) [inline src/script/config/dependenciesGraph.js]: Why did you remove this?

Comment (bennyn) [reply src/script/config/dependenciesGraph.js]: Because it has static functions now and does not need to be resolved anymore.

File: src/script/cache/CacheRepository.ts [hunk 5]
```diff
@@ -17,30 +17,34 @@
...
 export class CacheRepository {
-  constructor(logger) {
-    this.logger = logger;
+  public static CACHE_KEY = {
+    // Redefine name from "sw.js"
+    ASSETS: `asset: 'asset-cache-v2`,
+  };
+
+  // Caution! When refactoring this code, make sure that it works in an "beforeunload" scenario!
+  static async clearCacheStorage(): Promise<boolean> {
```
Comment (atomrc) [inline src/script/cache/CacheRepository.ts:32]: What do you think of just a module that exports 2 functions instead of a class + static methods?
